src/java.desktop/macosx/native/libawt_lwawt/java2d/metal/MTLRenderer.m

```diff
@@ -76,10 +76,63 @@ void MTLRenderer_DrawLine(MTLContext *mtlc, BMTLSDOps * dstOps, jint x1, jint y1
     if (mtlEncoder == nil)
         return;
 
-    struct Vertex verts[2] = {
-            {{x1, y1}},
-            {{x2, y2}}
-    };
+    // DrawLine implementation same as in OGLRenderer.c
+    struct Vertex verts[2];
+    if (y1 == y2) {
+        // horizontal
+        float fx1 = (float)x1;
+        float fx2 = (float)x2;
+        float fy  = ((float)y1) + 0.2f;
+
+        if (x1 > x2) {
+            float t = fx1; fx1 = fx2; fx2 = t;
+        }
+
+        struct Vertex v1 = {{fx1 + 0.2f, fy}};
```

Comment:
We can avoid creation of v1 and v2 by directly assigning to verts[0] and verts[1].
8250778 changes can be referred.

Reply:
Fixed!